grabbit/src/test/groovy/com/twcable/grabbit/jcr/ProtoPropertyDecoratorSpec.groovy

```diff
@@ -0,0 +1,210 @@
+        given:
+        final mockNode = Mock(Node)
+
+        when:
+        PropertyProto someMultiProperty = PropertyProto.newBuilder()
+                                              .setName("somemultivalueproperty")
+                                              .setType(PropertyType.STRING)
+                                              .setValues(
+                                                  ProtoValues.newBuilder()
+                                                      .addAllValue(
+                                                          [
+                                                              ProtoValue.newBuilder().setStringValue("value1").build(),
+                                                              ProtoValue.newBuilder().setStringValue("value2").build()
+                                                          ]
+                                                      )
+                                                      .build()
+                                              )
+                                              .build()
+
+        final propertyDecorator = new ProtoPropertyDecorator(someMultiProperty)
+        propertyDecorator.writeToNode(mockNode)
+
+        then:
+        1 * mockNode.setProperty("somemultivalueproperty", [new StringValue("value1"), new StringValue("value2")] as Value[], PropertyType.STRING)
+    }
+
+
+    def "A single-value property that is attempted to be written to a multi-value property can recover, and be written"() {
+        given:
+        final mockNode = Mock(Node)
+
+        when:
+        PropertyProto someProperty = PropertyProto.newBuilder()
+                                              .setName("somename")
+                                              .setType(PropertyType.STRING)
+                                              .setValue(
+                                                  ProtoValue.newBuilder()
+                                                      .setStringValue("somevalue")
+                                                      .build()
+                                              )
+                                              .build()
+
+        mockNode.setProperty("somename", new StringValue("somevalue"), PropertyType.STRING) >> { throw new ValueFormatException("Multivalued property can not be set to a single value") }
+        mockNode.getProperty("somename") >> {
+            Mock(Property) {
+                isMultiple() >> true
+            }
+        }
+
+        final propertyDecorator = new ProtoPropertyDecorator(someProperty)
+        propertyDecorator.writeToNode(mockNode)
+
+        then:
+        notThrown(ValueFormatException)
+        1 * mockNode.setProperty("somename", [new StringValue("somevalue")] as Value[], PropertyType.STRING)
+    }
+
+
+    def "A multi-value property that is attempted to be written to a single-value property can recover, and be written"() {
```

Comment:
Did you do something with setup() before you ran these tests?  Could you try again?